src/listr/tasks/tenderdash/initTaskFactory.js

```diff
@@ -0,0 +1,58 @@
+const { Listr } = require('listr2');
+
+/**
+ * @param {initializeTenderdashNode} initializeTenderdashNode
+ * @param {Docker} docker
+ * @return tenderdashInitTask
+ */
+function tenderdashInitTaskFactory(
+  initializeTenderdashNode,
+  docker,
+) {
+  /**
+   * @param {Config} config
+   * @return {Listr}
+   */
+  function tenderdashInitTask(
+    config,
+  ) {
+    return new Listr([
+      {
+        title: 'Validate keys and volumes',
+        task: async () => {
+          const isValidatorKeyEmpty = Object.keys(config.get('platform.drive.tenderdash.validatorKey')).length === 0;
+          const isNodeKeyEmpty = Object.keys(config.get('platform.drive.tenderdash.nodeKey')).length === 0;
+          const isGenesisEmpty = Object.keys(config.get('platform.drive.tenderdash.genesis')).length === 0;
+
+          const { Volumes: existingVolumes } = await docker.listVolumes();
+          const { COMPOSE_PROJECT_NAME: composeProjectName } = config.toEnvs();
+          const isDataVolumeMissing = !existingVolumes.find((v) => v.Name === `${composeProjectName}_drive_tenderdash`);
+
+          if (isValidatorKeyEmpty || isNodeKeyEmpty || isNodeKeyEmpty || isDataVolumeMissing) {
```

Comment:
You can do `task.skip` and `return` if you meet the opposite condition.

Reply:
OK, I think I inverted the logic correctly and everything works. I'm not sure where I am supposed to put `return`, it just creates a lot of errors about "Unnecessary return statement." or "Expected to return a value at the end of async arrow function 'task'."